packages/contents/src/drivecontent.ts

```diff
@@ -0,0 +1,244 @@
+import { PathExt } from '@jupyterlab/coreutils';
+import { Contents } from '@jupyterlab/services';
+import { BLOCK_SIZE, TDriveMethod, TDriveRequest, TDriveResponse } from './drivefs';
+import { DIR_MODE, FILE_MODE } from './emscripten';
+
+export interface IDriveContentsProcessor {
+  /**
+   * Process a content request
+   *
+   * @param request the request
+   */
+  processDriveRequest<T extends TDriveMethod>(
+    request: TDriveRequest<T>,
+  ): Promise<TDriveResponse<T>>;
+
+  /**
+   * Process the request to read a directory content
+   *
+   * @param request the request
+   */
+  readdir(request: TDriveRequest<'readdir'>): Promise<TDriveResponse<'readdir'>>;
+
+  /**
+   * Process the request to remove a directory
+   *
+   * @param request the request
+   */
+  rmdir(request: TDriveRequest<'rmdir'>): Promise<TDriveResponse<'rmdir'>>;
+
+  /**
+   * Process the request to rename a file or directory
+   *
+   * @param request the request
+   */
+  rename(request: TDriveRequest<'rename'>): Promise<TDriveResponse<'rename'>>;
+
+  /**
+   * Process the request to get the node mode (file or directory)
+   *
+   * @param request the request
+   */
+  getmode(request: TDriveRequest<'getmode'>): Promise<TDriveResponse<'getmode'>>;
+
+  /**
+   * Process the request to check if a node exist
+   *
+   * @param request the request
+   */
+  lookup(request: TDriveRequest<'lookup'>): Promise<TDriveResponse<'lookup'>>;
+
+  /**
+   * Process the request to create a directory/file
+   *
+   * @param request the request
+   */
+  mknod(request: TDriveRequest<'mknod'>): Promise<TDriveResponse<'mknod'>>;
+
+  /**
+   * Process the request to get a node stats
+   *
+   * @param request the request
+   */
+  getattr(request: TDriveRequest<'getattr'>): Promise<TDriveResponse<'getattr'>>;
+
+  /**
+   * Process the request to get the content of a file
+   *
+   * @param request the request
+   */
+  get(request: TDriveRequest<'get'>): Promise<TDriveResponse<'get'>>;
+
+  /**
+   * Process the request to write the content of a file
+   *
+   * @param request the request
+   */
+  put(request: TDriveRequest<'put'>): Promise<TDriveResponse<'put'>>;
+}
+
+/**
+ * Class for processing a drive request from the DriveFS.
+ */
+export class DriveContentsProcessor implements IDriveContentsProcessor {
+  private contentsManager: Contents.IManager;
+
+  constructor(options: DriveContentsProcessor.IOptions) {
+    this.contentsManager = options.contentsManager;
+  }
+
+  async processDriveRequest<T extends TDriveMethod>(
+    request: TDriveRequest<T>,
+  ): Promise<TDriveResponse<T>> {
+    switch (request.method) {
+      case 'readdir':
+        return this.readdir(request as TDriveRequest<'readdir'>) as Promise<
```

Comment:
I understand we have no choice to cast here. It seems you're casting to `any` in https://github.com/jupyter-lsp/jupyterlab-lsp/pull/278/files#diff-7c7804a4b99d4b67b07ab518d033a6de85bdbf291e5e03400106901ff2501d5aR30.

Reply:
From outside calls of `processDriveRequest`, the response type is correctly computed.